<a href="https://colab.research.google.com/github/justin-hsieh/DS-Unit-1-Sprint-3-Statistical-Tests-and-Experiments/blob/master/LS_DS_131_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 3 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

Stretch goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Apply hypothesis testing to your personal project data (for the purposes of this notebook you can type a summary of the hypothesis you formed and tested)

In [0]:
import pandas as pd
import numpy as np
from scipy.stats import ttest_1samp
from scipy.stats import ttest_ind, ttest_ind_from_stats, ttest_rel

In [3]:
df = pd.read_csv('house-votes-84.data')
df.columns = ['Class Name', 'Handicapped_Infants','Water_project_cost_sharing',
              'Adoption_of_the_budget_res',
              'Physician_fee_freeze','El_Salvador_aid', 'Religious_Groups_in_schools',
              'Anti_Satellite_test_ban','Aid_to_Nicaraguan_contras','Mx_missile',
              'Immigration','Synfuels_corporation_cutback','Education_spending',
              'Superfund_right_to_sue','Crime','Duty_free_exports','export_Admin_act_SA']
df.replace({'y':  1, 'n': 0,'?': np.nan}, inplace=True)
df.head()

Class Name  Handicapped_Infants  ...  Duty_free_exports  export_Admin_act_SA
0  republican                  0.0  ...                0.0                  NaN
1    democrat                  NaN  ...                0.0                  0.0
2    democrat                  0.0  ...                0.0                  1.0
3    democrat                  1.0  ...                1.0                  1.0
4    democrat                  0.0  ...                1.0                  1.0

[5 rows x 17 columns]

In [4]:
dems = df[df['Class Name'] == 'democrat']
dems.head()

Class Name  Handicapped_Infants  ...  Duty_free_exports  export_Admin_act_SA
1   democrat                  NaN  ...                0.0                  0.0
2   democrat                  0.0  ...                0.0                  1.0
3   democrat                  1.0  ...                1.0                  1.0
4   democrat                  0.0  ...                1.0                  1.0
5   democrat                  0.0  ...                1.0                  1.0

[5 rows x 17 columns]

In [5]:
reps = df[df['Class Name'] == 'republican']
reps.head()

Class Name  Handicapped_Infants  ...  Duty_free_exports  export_Admin_act_SA
0   republican                  0.0  ...                0.0                  NaN
6   republican                  0.0  ...                NaN                  1.0
7   republican                  0.0  ...                0.0                  1.0
9   republican                  0.0  ...                0.0                  0.0
10  republican                  0.0  ...                NaN                  NaN

[5 rows x 17 columns]

##1-Sample t-test Example



In [23]:
dems['Handicapped_Infants'].value_counts(dropna=True)

1    156
0    102
Name: Handicapped_Infants, dtype: int64

In [27]:
dems['Handicapped_Infants'].mean()

0.6046511627906976

In [0]:
ttest_1samp(dems['Handicapped_Infants'], 0.5, nan_policy='omit')

In [0]:
ttest_1samp(dems['Handicapped_Infants'], 0, nan_policy='omit')

In [0]:
ttest_1samp(dems['Handicapped_Infants'], 1, nan_policy='omit')

In [0]:
ttest_1samp(dems['Handicapped_Infants'], 0.6, nan_policy='omit')

##2-Sample t-test Example

In [0]:
print('Democratic Support: ', dems['export_Admin_act_SA'].mean())
print('Republican Support: ', reps['export_Admin_act_SA'].dropna().mean())

In [41]:
ttest_ind(dems['export_Admin_act_SA'], reps['export_Admin_act_SA'], nan_policy='omit')

Ttest_indResult(statistic=6.893205322931943, pvalue=2.8060959129817107e-11)

#Assignment

### 2.
Null Hypothesis: The number of votes are the same for both sides

Alternate Hypothesis: Thenumber of votes are different

In [6]:
print('Democratic Support: ', dems['Synfuels_corporation_cutback'].mean())
print('Republican Support: ', reps['Synfuels_corporation_cutback'].mean())

Democratic Support:  0.5058823529411764
Republican Support:  0.1320754716981132


In [7]:
ttest_ind(dems['Synfuels_corporation_cutback'], reps['Synfuels_corporation_cutback'], nan_policy='omit')

Ttest_indResult(statistic=8.293603989407588, pvalue=1.5759322301054064e-15)

Conclusion: With a t-stat of 8.293603989407588 and a p-value of 1.5759322301054064e-15, the null hpyothesis is rejected

### 3.
Null Hypothesis: The number of votes are the same for both sides

Alternate Hypothesis: Thenumber of votes are different

In [8]:
print('Democratic Support: ', dems['El_Salvador_aid'].mean())
print('Republican Support: ', reps['El_Salvador_aid'].mean())

Democratic Support:  0.21568627450980393
Republican Support:  0.9512195121951219


In [9]:
ttest_ind(dems['El_Salvador_aid'], reps['El_Salvador_aid'], nan_policy='omit')

Ttest_indResult(statistic=-21.06444879561732, pvalue=1.3035583587179729e-67)

Conclusion: With a t-stat of -21.06444879561732 and a p-value of 1.3035583587179729e-67, the null hpyothesis is rejected

###4.
Null Hypothesis: The number of votes are the same for both sides

Alternate Hypothesis: Thenumber of votes are different

In [25]:
print('Democratic Support: ', dems['Water_project_cost_sharing'].mean())
print('Republican Support: ', reps['Water_project_cost_sharing'].mean())

Democratic Support:  0.502092050209205
Republican Support:  0.5034013605442177


In [26]:
ttest_ind(dems['Water_project_cost_sharing'], reps['Water_project_cost_sharing'], nan_policy='omit')

Ttest_indResult(statistic=-0.02491808700047811, pvalue=0.9801332440121653)

Conclusion: With a t-stat of -0.02491808700047811 and a p-value of 0.9801332440121653, the null hpyothesis fails to be rejected

In [0]:
def support(Dcolumn1, Rcolumn2):
 
  return (print('Democratic Support: ',  Dcolumn1.mean()),
         print('Republican Support: ',Rcolumn2.mean()),
         ttest_ind(Dcolumn1, Rcolumn2, nan_policy='omit'))


In [44]:
support(dems['Water_project_cost_sharing'],reps['Water_project_cost_sharing'] )

Democratic Support:  0.502092050209205
Republican Support:  0.5034013605442177


(None,
 None,
 Ttest_indResult(statistic=-0.02491808700047811, pvalue=0.9801332440121653))